src/types/list.rs

```diff
@@ -29,14 +29,14 @@ pub(crate) fn new_from_iter<'py>(
     py: Python<'py>,
     elements: &mut dyn ExactSizeIterator<Item = PyObject>,
 ) -> Bound<'py, PyList> {
-    try_new_from_iter(py, &mut elements.map(Ok)).unwrap()
+    try_new_from_iter(py, &mut elements.map(|e| e.into_bound(py)).map(Ok)).unwrap()
 }
 
 #[inline]
 #[track_caller]
 pub(crate) fn try_new_from_iter<'py>(
     py: Python<'py>,
-    elements: &mut dyn ExactSizeIterator<Item = PyResult<PyObject>>,
+    elements: &mut dyn ExactSizeIterator<Item = PyResult<Bound<'py, PyAny>>>,
```

Comment:
Do you think it's worth making these have a concrete iterator? I forget why we had to change it for sets 🤔

Reply:
Do you mean `impl` instead of `dyn` here? Or what concrete type are you referring to?

Reply:
Yes, `impl Iterator`